### ISSUE TITLE
Testing elm-mdl

### ISSUE THREAD

**Issue by hakonrossebo** [debois/elm-mdl#227]
We need to agree on how we do testing in elm-mdl.

Currently, I'v added the "elm-test" skeleton, basically to be able to have something running and exiting with 0/1 in the scripts tag in travis.yml.

I guess testing with elm-test would be beneficial if we could be able to check/inspect the html returned from elm-mdl functions - then we could write unit tests/TDD to if verify if css classes has been set or not. I'm not sure if this is possible in any way, comments would be nice on this.

The other testing alternative is by using [https://github.com/lorenzo/elm-webdriver](https://github.com/lorenzo/elm-webdriver) - This would be able to inspect the correct css/markup on runtime.

This is what I'v described in TESTING.md for the current test implementation:

Tests are located in a separate *"tests"* folder and does not affect elm-mdl.

Prerequisite: To be able to run tests from your terminal - install node-test-runner by running:
> npm install -g elm-test

Run all tests by executing this command in the root folder:
>elm-test

The travis.yml file has been configured to run elm-test when any branch is pushed to GitHub

**NOTE**

One primary reason to include elm-test now, is to improve the speed of Travis build.
Currently, there are no actual tests implemented. The Material components and Demo components are imported by the testrunner and only verifies that building those modules does not fail.

We should look at other projects to see how to implement tests in Elm-mdl.
- [elm-test](https://github.com/elm-community/elm-test) - The elm-test package used
- [elm-css](https://github.com/rtfeldman/elm-css) - A relevant project using elm-test.
- [elm-combine](https://github.com/Bogdanp/elm-combine) - Another relevant project using elm-test.
- [Caching elm builds on Travis](https://8thlight.com/blog/rob-looby/2016/04/07/caching-elm-builds-on-travis-ci.html)
- [Article on testing in Elm](https://medium.com/@_rchaves_/testing-in-elm-93ad05ee1832#.tr0bpt18s) - Some parts are outdated.

**Comment by vipentti** [comment]
I think testing in general would be a thing to have. Especially as the library is fairly large and Mdl v2 is coming sometime in the future so having tests to help avoid regressions would be nice. 

On how to actually test elm-mdl, is another story. I think the most useful testing would be with something like elm-webdriver verifying that the correct css and markup is being applied correctly in specific situations, such as focus and blur events. It would be even better if there was a way to have it run on various browser engines (looking at you Firefox).
